File: Java/프로그래머스/2/60057. 문자열 압축/문자열 압축.java
import java.util.*;
class Solution {
    public int solution(String s) {
        int answer = s.length();

        for (int i = 1; i <= s.length() / 2; i++) {
            int len = 0;
            for (int j = 0; j + i <= s.length(); j += i) {
                int count = 1;
                int next = j + i;
                String sub = s.substring(j, j + i);
                while (next + i <= s.length() && sub.equals(s.substring(next, next + i))) {
                    count++;
                    next += i;
                }
                if (count == 1) len += i;
                else len += i + String.valueOf(count).length();
                j = next - i;
            }
            len += s.length() % i;
            answer = Math.min(answer, len);
        }

        return answer;
    }
}

File: Python/프로그래머스/2/60057. 문자열 압축/문자열 압축.py
def solution(s):
    answer = len(s)

    for i in range(1, len(s) // 2 + 1):
        length = 0
        j = 0

        while j + i <= len(s):
            count = 1
            next_idx = j + i
            sub_str = s[j : j + i]

            while next_idx + i <= len(s) and sub_str == s[next_idx : next_idx + i]:
                count += 1
                next_idx += i

            if count == 1:
                length += i
            else:
                length += i + len(str(count))

            j = next_idx

        length += len(s) % i
        answer = min(answer, length)

    return answer